### PR TITLE
Dataload fix

### PR DESCRIPTION
**Pull request by jahatef**
In #958 

https://github.com/EleutherAI/gpt-neox/pull/979/files#diff-e2b248f8c422a601bcb0b7d93f96c1dff070f2694737e2b69f1def64ab9c1844L276

When `data = None`, `_keys = ["text", "label"]`

This led to access errors:

https://github.com/EleutherAI/gpt-neox/pull/979/files#diff-e2b248f8c422a601bcb0b7d93f96c1dff070f2694737e2b69f1def64ab9c1844R280-R287

where `data_b` does not have any values in the `"label"` key.

Instead, 

https://github.com/EleutherAI/gpt-neox/pull/979/files#diff-e2b248f8c422a601bcb0b7d93f96c1dff070f2694737e2b69f1def64ab9c1844R306

was placed to only access the `"label"` key if `neox_args.label_data_paths` is set from the config file, which is the config determining if a labeled dataset is used.

### CODE REVIEW

File: megatron/training.py [hunk 1]
```diff
@@ -273,8 +273,7 @@ def pretrain(neox_args):
 
 def _get_batch(neox_args, tokenizer, keys, data, datatype):
     """Support function for get_batch / get_batch pipe (to avoid code repetition)"""
-    _keys = [k for k in keys if k in data] if data else keys
```
Comment (jahatef) [inline megatron/training.py:276]: if 'data  = None', this will set `keys = ["text", "label".

File: configs/1-3B.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@
    # parallelism settings ( you will want to change these based on your cluster setup, ideally scheduling pipeline stages
    # across the node boundaries )
    "pipe_parallel_size": 1,
-   "model_parallel_size": 1,
+   "model_parallel_size": 2,
```
Comment (Quentin-Anthony) [inline configs/1-3B.yml]: Don't make config changes

File: data/enwik8/enwik8.zip
Comment (Quentin-Anthony) [inline data/enwik8/enwik8.zip:1]: remove this change from the PR. We don't want `enwik8.zip` to be hosted on gh

File: megatron/data/data_utils.py [hunk 3]
```diff
@@ -67,6 +67,7 @@ def build_the_dataset(
     indexed_dataset = make_indexed_dataset(data_prefix, data_impl, skip_warmup)
     if label_prefix is None:
         label_dataset = None
+
```
Comment (Quentin-Anthony) [inline megatron/data/data_utils.py:70]: remove this space change

File: megatron/training.py [hunk 4]
```diff
@@ -324,12 +323,12 @@ def get_batch(neox_args, data_iterator):
...
     """A modification of get_batch() to work with the latest batch instead of an iterator."""
     # Items and their type.
-    keys = ["text", "label"]
+    keys = ["text", "label"] if neox_args.label_data_paths else ["text"]
     datatype = torch.int64
 
     tokens, labels, loss_mask, attention_mask, position_ids = _get_batch(
         neox_args, neox_args.tokenizer, keys, data, datatype
-    )
+    )
```
Comment (Quentin-Anthony) [inline megatron/training.py:331]: remove this space from the pr

**Comment by StellaAthena** [comment]
@Quentin-Anthony I have made the requested changes. I'm finding that training hangs in mulit-node settings with tp > 1, as reported in #985. However I see the same on `main` and don't think this PR is to blame.

**Review by StellaAthena — APPROVED** [review]
After applying this fix to _both nodes_ it now works. Whoops xD
